### ISSUE TITLE
Tokenizer update

### ISSUE THREAD

**Issue by scottkleinman** [WheatonCS/Lexos#377]
The Tokenizer page should be functional on your next pull. A few options are still not working properly, and they have been disabled for the moment. But you should be able to use most of the options.

**Comment by mleblanc321** [comment]
nice;  we want class labels :)   (topWords folks could use 'em for testing)

**Comment by scottkleinman** [comment]
OK. I'll switch to that.

**Comment by mleblanc321** [comment]
:)

**Comment by scottkleinman** [comment]
The save function was commented out. I've uncommented it and pushed the change, so class labels should work.

Note that there seems to be a problem updating changes on the page. I recommend refreshing the page to show the results of the most recent actions until I can sort out why this is happening.

**Comment by YiqingC** [comment]
Tokenizer still has issues:

(1) the button is not functional at all;  we get a table loaded on first load, but the button never updates the table

![tokenizer_error_message](https://cloud.githubusercontent.com/assets/15393424/16432484/2969160c-3d53-11e6-8422-c911f311c919.png)



(2) on right-click of a document in a Table row, when i select Preview (i see the modal appear with text) but then after i close the Preview window, this particular row is DE-selected

(3) Sort by Value toggle hangs page  (if i click sort increasing/decreasing arrows)

**Comment by scottkleinman** [comment]
Bugs 1 and 3 are due to a change Alvaro made on the back end. They should go away when he resolves the error.

Bug 2 sounds like it might be for Manage, as there is no right-clicking in Tokenizer. I am unable to reproduce it. Can you give me more information?

**Comment by mleblanc321** [comment]
crap, right;  Bug 2 is on Manage (sorry, mixing up pages) ... hmm, now I am not able to replicate Bug 2

**Comment by Landaluce** [comment]
I fixed the back end error, I just pushed my changes

**Comment by Landaluce** [comment]
but the Regenerate Table button still not functional

**Comment by scottkleinman** [comment]
The Regenerate Table button works for everything but assigning temporary labels and rotating the table. However, the changes don't show unless click on a function within the table, such as changing the page.

**Comment by evanlaferriere** [comment]
The regenerate button is not working for me at all

**Comment by scottkleinman** [comment]
I'm running out of steam for the night, so I'm going push what I've got and make a report here. I'm pretty sure that the reason the Regenerate button wasn't working was because the javascript was broken by a reference to an undefined variable for the temporary labels. I uncommented that line, and the button seemed to work again.

That said, dealing with the temporary labels is proving to be phenomenally difficult. As a fall back, we can simply omit the function. If user really want it, they can change the document names in Manage. But, having said that, I have been able to do it, albeit in a very torturous way.

At the moment, I'm simply destroying and re-drawing the table every time the button is clicked. I'm then serialising the form values and testing each one for a key beginning with "file_". I then use the matches to construct the table header string, superseding the one returned from the ajax response.

This seems to work. The one problem I am encountering is that the columns then appear in sequential order, rather than alphabetical order (the default behaviour). It should be possible to introduce a sorting mechanism to handle this, but I'm out of steam for tonight. So this is mainly to give everyone an update and to server as a reminder to me of where I left off.
